pkg/store/proxy.go

```diff
@@ -269,10 +310,40 @@ func startStreamSeriesSet(
 	return s
 }
 
-// Next blocks until new message is received or stream is closed.
+// Next blocks until new message is received or stream is closed or operation is timed out.
 func (s *streamSeriesSet) Next() (ok bool) {
-	s.currSeries, ok = <-s.recvCh
-	return ok
+	//Some times GRPC streams get stuck and don't send any data for long periods of time
```

Comment:
so this is misleading. It's not gRPC streams being stuck, just StoreAPI server is stuck (: